### PR TITLE
 🌱 Refactor cluster permission

### PR DESCRIPTION
**Pull request by qiujian16**
## Summary

This is to use lister for permission to avoid too frequent get calls. 
For all legacy k8s resource managed by registration, we add a cluster label and informer will use label selector to filter.

## Related issue(s)

Fixes #

### CODE REVIEW

**Comment by skeeey** [comment]
LGTM

File: pkg/common/apply/rbac.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+			result.Result, result.Changed, result.Error = Apply[*rbacv1.ClusterRole](
+				ctx, m.clusterRoleLister, m.client.RbacV1().ClusterRoles(), compareClusterRole, t, recorder)
+		case *rbacv1.ClusterRoleBinding:
+			result.Result, result.Changed, result.Error = Apply[*rbacv1.ClusterRoleBinding](
+				ctx, m.clusterRoleBindingLister, m.client.RbacV1().ClusterRoleBindings(), compareClusterRoleBinding, t, recorder)
+		case *rbacv1.Role:
+			result.Result, result.Changed, result.Error = Apply[*rbacv1.Role](
+				ctx, m.roleLister.Roles(t.Namespace), m.client.RbacV1().Roles(t.Namespace), compareRole, t, recorder)
+		case *rbacv1.RoleBinding:
+			result.Result, result.Changed, result.Error = Apply[*rbacv1.RoleBinding](
```
Comment (zhujian7) [inline pkg/common/apply/rbac.go:72]: if the "roleRef" field changes, the apply will fail, right?

Comment (qiujian16) [reply pkg/common/apply/rbac.go:73]: maybe not for now. it is implicit

**Pull request merged**